src/app/frontend/deploy/deployfromsettings.html

```diff
@@ -17,17 +17,19 @@
 <kd-help-section>
   <md-input-container class="md-block" md-is-error="ctrl.isNameError()">
     <label>App name</label>
-    <input ng-model="ctrl.name" name="name" namespace="ctrl.namespace" required kd-unique-name
-        ng-model-options="{ updateOn: 'default blur', debounce: { 'default': 500, 'blur': 0 } }">
-    <md-progress-linear class="kd-deploy-form-progress" md-mode="indeterminate"
-        ng-show="ctrl.form.name.$pending">
-    </md-progress-linear>
-    <ng-messages for="ctrl.form.name.$error" role="alert" multiple>
-      <ng-message when="required">Application name is required.</ng-message>
-      <ng-message when="uniqueName">
-        Application with this name already exists within namespace <i>{{ctrl.namespace}}</i>.
-      </ng-message>
-    </ng-messages>
+    <input ng-model="ctrl.name"
+           name="name"
+           namespace="ctrl.namespace"
+           required
+           kd-unique-name
+           ng-pattern="/^[a-z]+[a-z\-]*[a-z]*$/">
+
+    <div ng-messages="ctrl.form.name.$error" ng-if="ctrl.form.name.$touched">
```

Comment:
@digitalfishpond Thanks for through investigation. I agree with @cheld that this is Angular Material bug. It shouldn't behave like this. Having said that, I don't think we should fix this. To me, the best way is just to wait until Angular Material fixes it. 
What do you think?